test-framework/junit5/src/main/java/org/apache/camel/quarkus/test/TestConfigSourceFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.test;
+
+import java.util.Collections;
+import java.util.HashMap;
+
+import io.smallrye.config.ConfigSourceContext;
+import io.smallrye.config.ConfigSourceFactory;
+import io.smallrye.config.common.MapBackedConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+public class TestConfigSourceFactory implements ConfigSourceFactory {
```

Comment:
Thinking again about logging. We should remove this class.

Logging should be left to the user to configure. This configuration only applies to the internal testing of the Camel main project. See [here](https://github.com/apache/camel/blob/main/components/camel-test/camel-test-junit5/src/test/resources/log4j2.properties#L20-L22). Users are free to configure things however they wish in their application.

Reply:
@jamesnetherton  I agree, as a compromise I suggest to remove this class and add a example of configuration into the doc for the users, who would like to achieve the same output as is with default `CamelTestSupport`. Do you agree?

Reply:
> achieve the same output as is with default `CamelTestSupport`

How is the output in `CamelTestSupport` different on vanilla Camel to that of Camel Quarkus? Isn't the same log content being produced on all platforms from `CamelTestSupport`?

How the logs are formatted is governed by the log framework configuration. AFAIK, camel makes no assumptions about this.

I see it as more critical that the test user experience is the same as in Camel. The formatting of the logs is not so important IMO.

Or am I misunderstanding something?

Reply:
I'm sorry I overlooked the location of log configuration in `camel-test` module. My advice doesn't make sense, therefore the whole class as you suggested is pointless. (I thought that the camel-test-junit5 also defines default log configuration, therefore the class. Because it is not correct, the class shouldn't be part of the PR. It will be removed)

Reply:
done